astacus/coordinator/plugins/clickhouse/object_storage.py

```diff
@@ -116,16 +125,20 @@ def _copy_via_local_filesystem(
     stats: StatsClient | None,
 ) -> None:
     keys_to_copy = len(keys)
-    for keys_copied, key in enumerate(keys):
+    for keys_copied, key in enumerate(keys, start=1):
         with tempfile.TemporaryFile(dir=copy_config.temporary_directory) as temp_file:
             metadata = source.get_contents_to_fileobj(key, temp_file)
             target.store_file_object(key, temp_file, metadata)
         if stats:
-            stats.gauge(
-                "astacus_restore_clickhouse_tiered_storage_keys_remaining",
-                keys_to_copy - keys_copied,
-                tags={"copy_method": copy_config.method},
-            )
+            _handle_copy_progress_fn(keys_copied, keys_to_copy, copy_config.method, stats)
+
+
+def _handle_copy_progress_fn(completed_files: int, total_files: int, copy_method: str, stats: StatsClient) -> None:
```

Comment:
The stats client is usually the first arg. We also deviate from the typical style of systematically using underscores for helper functions.

Reply:
In this case, please make the args following the `stats` argument keywork only. IDEs usually complete the arg name, but reading `_handle_copy_progress_fn(stats, 1, 2, "mymethod")` out of context isn't very readable compared to `_handle_copy_progress_fn(stats, completed_files=1, total_files=2, copy_method="mymethod")`

Reply:
Thanks for the advise

- added * after `stats`
- moved stats before additional arguments

```python
_handle_copy_progress_fn(stats: StatsClient, *, completed_files: int, total_files: int, copy_method: str) -> None:
```

Expecting
```python
_handle_copy_progress_fn(stats, completed_files=1, total_files=2, copy_method="mymethod")
```